src/lib/builtins/mod.rs

```diff
@@ -1,3 +1,15 @@
+/// Macro to create a new member function of a prototype
+macro_rules! make_builtin_fn {
+    ($fn:ident, named $name:expr, with length $l:tt, of $p:ident) => {
+        let $fn = to_value($fn as NativeFunctionData);
+        $fn.set_field_slice("length", to_value($l));
+        $p.set_field_slice($name, $fn);
+    };
+    ($fn:ident, named $name:expr, of $p:ident) => {
+        $p.set_field_slice($name, to_value($fn as NativeFunctionData));
```

Comment:
Are there any functions that do not have their `length` set (e.g. it's `undefined`)? If so, this line is fine.

But if not, I think it would be better if you set the `length` to `0` by default.